polaris-react/src/components/ActionList/ActionList.tsx

```diff
@@ -99,16 +110,47 @@ export function ActionList({
       </>
     ) : null;
 
+  const totalActions =
+    finalSections?.reduce(
+      (acc: number, section) => acc + section.items.length,
+      0,
+    ) || 0;
+
+  const totalFilteredActions = useMemo(() => {
+    const totalSectionItems =
+      filteredSections?.reduce(
+        (acc: number, section) => acc + section.items.length,
+        0,
+      ) || 0;
+
+    return totalSectionItems;
+  }, [filteredSections]);
+
+  const showSearch = totalActions >= 10;
```

Comment:
I do like how this doesn't add a prop and then enforces this UX pattern consistently. This would affect current usages where total actions are greater than or equal to 10, though I'd bet that is a very low amount.